### PR TITLE
INFRA-9588: Fix spawning of public cloud instances with private interface

### PR DESCRIPTION
**Pull request by tortuegenialez**
- Add a module to retrieve private network and return the openstack id of the network matching a given region.
- Modify the module attaching an interface to a public cloud instance to use the openstack id.

### CODE REVIEW

File: plugins/modules/public_cloud_instance_interface.py [hunk 1]
```diff
@@ -76,21 +78,21 @@ def run_module():
...
-                         .format(interface_ip, interface_network_id, state, instance_id),
+                         .format(interface_ip, interface_openstack_id, state, instance_id),
                          changed=True)
 
     if state == 'absent':
         # Need to get the interface id (via /cloud/project/{serviceName}/instance/{instanceId}/interface).
         # How to manage multiple interfaces ?
         module.fail_json(msg="Removing an interface is not yet implemented")
     if state == 'present':
-        result = client.wrap_call("POST", f"/cloud/project/{service_name}/instance/{isinstance}/interface",
```
Comment (tortuegenialez) [inline plugins/modules/public_cloud_instance_interface.py:91]: `isinstance` is an error in the last PR. It should have been `instance_id` instead

File: plugins/modules/public_cloud_private_network_info.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    private_network:
+        required: true
+        description: The OVH private network
+    region:
+        required: true
+        description: The region where to lookup for network
+"""
+
+EXAMPLES = r"""
+  - name: Get openstackId
```
Comment (Skunnyk) [inline plugins/modules/public_cloud_private_network_info.py:35]: `Get openstackId of the network` or something like that

Comment (carrouet) [reply plugins/modules/public_cloud_private_network_info.py]: I would even say `Get openstackId of the network in this region` if its value is region specific

Comment (tortuegenialez) [reply plugins/modules/public_cloud_private_network_info.py]: Fixed, thaks!

**Review by Skunnyk — APPROVED** [review]

**Review by carrouet — APPROVED** [review]

**Pull request merged**